src/ActionsMenuItem.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from 'react';
+
+import { injectIntl } from 'react-intl';
+import type { IntlShape } from 'react-intl';
+
+import './ActionsMenuItem.scss';
+
+type ActionsMenuItemProps = {
+    intl: IntlShape,
+    isAllowed?: boolean,
+    isUsed?: boolean,
+    itemKey: string,
+    onChange: (event: Event) => void
+}
+
+export class ActionsMenuItem extends React.Component< ActionsMenuItemProps, {} > {
+    render () {
+        // We don't use FormattedMessage as we are working with a complex chain of templates.
+        const commandName = this.props.intl.formatMessage({ id: `Command.${this.props.itemKey}` });
+        const commandNameShort = this.props.intl.formatMessage({ id: `Command.short.${this.props.itemKey}` });
+
+        const actionNameKey = this.props.isAllowed ? "ActionsMenu.item.action.show" : "ActionsMenu.item.action.hide";
+        const actionName = this.props.intl.formatMessage({ id: actionNameKey });
+
+        // If we're used, show one message. If we're not, show another that differs based on `isAllowed`.
+        const showHideLabelKey = this.props.isUsed ? "ActionsMenu.item.usedItemToggleLabel" : "ActionsMenu.item.unusedItemToggleLabel";
+        const showHideLabel = this.props.intl.formatMessage(
+            { id: showHideLabelKey },
+            { action: actionName, commandName: commandName }
+        );
+
+        const showHideAriaLabelKey = this.props.isUsed ? "ActionsMenu.item.usedItemToggleAriaLabel" : "ActionsMenu.item.unusedItemToggleAriaLabel";
+        const showHideAriaLabel = this.props.intl.formatMessage(
+            { id: showHideAriaLabelKey },
+            { action: actionName, commandName: commandName }
+        );
+
+        return (
+            <div className="ActionsMenuItem">
+                <div className={'ActionsMenuItem__text' + (this.props.isAllowed ? '' : ' ActionsMenuItem__text--disabled')}>
+                    {commandNameShort}
+                </div>
+                <button className="ActionsMenuItem__button" aria-label={showHideAriaLabel} onClick={this.props.onChange}>
```

Comment:
Currently, there are 2 focusable items doing the checking of the items in the menu, which causes more keyboard interactions to go down through the menu. We could keep onClick handler on this button component to give more target spaces than a checkbox, and give tabIndex=-1 to the checkboxes to make checkboxes not focusable.

Reply:
Good idea.

Reply:
Hey Tony, I realized that my suggestion has an accessibility issue; because we are using button as a focusable item, screen readers don't have any information about whether the checkbox is checked or not. So I looked into alternative implementations, and figured out that if we provide label component for each checkbox item, that helps to increase click target size by allowing interaction with checkboxes by clicking the labels as well. I will leave some comments to this approach.